### DISCUSSION TITLE
What happens to return string pointer from file handler

### DISCUSSION THREAD

**Discussion by OsakiTsukiko** [webui-dev/webui#521] [General]
I am using webui-zig and the signature for the file handler is `fn my_files_handler(filename: []const u8) ?[]u8` and it seems to not be freed. Why is the return value not freed afterward? There is little purpose in having a filehandle that only returns static strings. Also, I can not see how this works in C, wouldn't local variables go out of scope? How do you send a char pointer to a locally defined variable?

**Comment by hassandraga** [comment]
Thank you for pointing this out. It seems the Zig example is missing some important comments. I will update it soon.

However, if you allocate memory using `webui.malloc()` then webui will take care of freeing this memory after sending data to the browser. You can check this [C example's](https://github.com/webui-dev/webui/blob/12dee8bab7d8a1817fc2535f8407b628a5a15e34/examples/C/serve_a_folder/main.c#L110) comments.

**Zig Example:**
```zig
fn my_files_handler(filename: []const u8) ?[]const u8 {
        const buffer = webui.malloc(1024 * 4);
        return buffer; // WebUI will automatically free buffer
}
```

**Reply by hassandraga** [reply]
Added (https://github.com/webui-dev/zig-webui/commit/c571a0b6e8276176d106eaf544b84291c6996b5c).

**Reply by OsakiTsukiko** [reply]
oh, that's nice, thanks!
